src/init.cpp

```diff
@@ -155,6 +155,9 @@ struct parseFromCommandLine {
 template <class T>
 void updateConfigurationValue(const pika::program_options::variables_map& vm, T& var,
                               const std::string& env_var, const std::string& cmdline_option) {
+  DLAF_ASSERT(env_var.find("DLAF") == std::string::npos, env_var);
+  DLAF_ASSERT(cmdline_option.find("dlaf") == std::string::npos, cmdline_option);
```

Comment:
I made these as conservative as possible at the moment, meaning `dlaf`/`DLAF` isn't allowed _anywhere_ in the strings. This is not meant to be a hard rule for the future, it only reflects the current state. If there's a need for having `DLAF`/`dlaf` somewhere in the names in the future we can relax this to e.g. only checking for prefixes.